src/typing_extensions.py

```diff
@@ -864,17 +847,9 @@ def __round__(self, ndigits: int = 0) -> T_co:
 
 
 def _ensure_subclassable(mro_entries):
-    def inner(func):
-        if sys.implementation.name == "pypy" and sys.version_info < (3, 9):
-            cls_dict = {
-                "__call__": staticmethod(func),
-                "__mro_entries__": staticmethod(mro_entries)
-            }
-            t = type(func.__name__, (), cls_dict)
-            return functools.update_wrapper(t(), func)
-        else:
-            func.__mro_entries__ = mro_entries
-            return func
+    def inner(obj):
+        obj.__mro_entries__ = mro_entries
+        return obj
     return inner
```

Comment:
We _could_ get rid of this `_ensure_subclassable` decorator altogether now, and monkey-patch `__mro_entries__` onto the `TypedDict` and `NamedTuple` functions manually (the same as CPython does, and the same as we used to do before https://github.com/python/typing_extensions/commit/dcdc53f62ecbea8a0fc9a42f66746f970ab101bb).

But aesthetically I sort-of prefer it as a decorator, so I'm happy to keep it the way you have it currently!